### PR TITLE
test(e2e): pod remove test fixes

### PR DESCRIPTION
**Pull request by chriswldenyer**
Remove test storage class unconditionally.
Better removal of timed-out created objects.
Remove dependency on number of cluster nodes and use
only 2 replicas for the test.
Remove non-ci local-cluster tests.
Use logf to output progress.

### CODE REVIEW

**Comment by chriswldenyer** [comment]
bors try

**Review by paulyoong — APPROVED** [review]

**Comment by chriswldenyer** [comment]
bors try

**Review by arne-rusek — CHANGES_REQUESTED** [review]

File: test/e2e/node_disconnect/lib/node_disconnect_lib.go [hunk 1]
```diff
@@ -164,6 +164,8 @@ func (env *DisconnectEnv) LossWhenIdleTest() {
 
 // Run fio against the cluster while a replica mayastor pod is unscheduled and then rescheduled
 func (env *DisconnectEnv) PodLossTest() {
+	Expect(len(env.otherNodes)).To(BeNumerically("==", 2)) // must run on a 3 node cluster
```
[the remark this answers is not in the document]
Comment (chriswldenyer) [reply test/e2e/node_disconnect/lib/node_disconnect_lib.go:167]: It could be done. I'll have a look.

**Comment by chriswldenyer** [comment]
Wii re-open as new PR because there are a lot of new changes.

**Pull request closed without merging**